sdk/storage/Azure.Storage.DataMovement/src/Shared/TransferManager.cs

```diff
@@ -292,6 +292,50 @@ public virtual async Task PauseTransferIfRunningAsync(string transferId, Cancell
             }
         }
 
+        /// <summary>
+        /// Resumes a transfer that has been paused or is in a completed state with failed or skipped transfers.
+        /// </summary>
+        /// <param name="transferId">The transfer ID of the transfer attempting to be resumed.</param>
+        /// <param name="sourceResource">A <see cref="StorageResource"/> representing the source.</param>
+        /// <param name="destinationResource">A <see cref="StorageResource"/> representing the destination.</param>
+        /// <param name="transferOptions">Options specific to this transfer.</param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>Returns a <see cref="DataTransfer"/> for tracking this transfer.</returns>
+        public virtual async Task<DataTransfer> ResumeTransferAsync(
+            string transferId,
+            StorageResource sourceResource,
+            StorageResource destinationResource,
+            TransferOptions transferOptions = default,
+            CancellationToken cancellationToken = default)
+        {
+            CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
+
+            // Attempt to add existing job to the checkpointer.
+            await _checkpointer.AddExistingJobAsync(
```

Comment:
If this is a resume, isn't the job already in the checkpointer (theoretically)?

Reply:
Hm I think you're right. When I initially implemented the resume workflow path, this was necessary cause we weren't initializing the entire checkpointer until we were asked to resume a specific transfer.

But after the change and addition I made [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.DataMovement/src/Shared/LocalTransferCheckpointer.cs#L52) to support `GetTransfers`, I probably don't need this any more?

Let me run a few quick tests to make sure this is the case before I remove it.